p2p/raintree/addrbook_utils.go

```diff
@@ -20,109 +15,62 @@ const (
 	floatPrecision            = float64(0.0000001)
 )
 
-// Whenever `addrBook` changes, we also need to update `addrBookMap` and `addrList`
-func (n *rainTreeNetwork) processAddrBookUpdates() error {
-	// OPTIMIZE(olshansky): This is a very naive approach for now that recomputes everything every time that we can optimize later
-	n.addrBookMap = make(map[string]*typesP2P.NetworkPeer, len(n.addrBook))
-	n.addrList = make([]string, len(n.addrBook))
-	for i, peer := range n.addrBook {
-		addr := peer.Address.String()
-		n.addrList[i] = addr
-		n.addrBookMap[addr] = peer
-	}
-	n.maxNumLevels = n.getMaxAddrBookLevels()
-
-	sort.Strings(n.addrList)
-	if i, ok := n.getSelfIndexInAddrBook(); ok {
-		// The list is sorted lexicographically above, but is reformatted below so this addr of this node
-		// is always the first in the list. This makes RainTree propagation easier to compute and interpret.
-		n.addrList = append(n.addrList[i:len(n.addrList)], n.addrList[0:i]...)
-	} else {
-		return fmt.Errorf("self address not found for %s in addrBook so this client can send messages but does not propagate them", n.selfAddr)
-	}
-	return nil
+// TODO(team): Need to integrate with persistence layer so we are storing this on a per height basis.
+// We can easily hit an issue where we are propagating a message from an older height (e.g. before
+// the addr book was updated), but we're using `maxNumLevels` associated with the number of
+// validators at the current height.
+func (n *rainTreeNetwork) getAddrBookLength(level uint32, _height uint64) int {
```

Comment:
NOOP: I like this! `_height`

Reminds me of elixir :)

Reply:
Reminds me of .NET few years ago. :)